recipes-openxt/vhd-scripts/vhd-scripts/vhd-copy

```diff
@@ -84,7 +84,7 @@ trap atexit EXIT
 
 size=$(vhd-util query -n $srcvhd -v)
 [ $? -eq 0 ] || error couldn\'t query size of src $srcvhd
-srctap=$(TAPDISK2_CRYPTO_KEYDIR="$srckeydir" TAPDISK3_CRYPTO_KEYDIR="$srckeydir" tap-ctl create -a "vhd:$srcvhd")
+srctap=$(TAPDISK2_CRYPTO_KEYDIR="$srckeydir" TAPDISK3_CRYPTO_KEYDIR="$srckeydir" tap-ctl create -R -a "vhd:$srcvhd")
```

Comment:
Are these `-R`'s needed if the patch to tap-ctl.c directly initializes flags to readonly?

Reply:
The -R's are not needed, but I added them to be explicit (and that was mentioned in the change log).

I don't like adding a blktap patch to implement this, but I don't know how else to do it.  It isn't strictly necessary, but it is a little safer to default to read-only.

Reply:
Claiming a VHD is read-only and measured but not measuring the metadata is "fine", but what this change does is strengthen the read-only+measured assertion to extend to the VHD metadata.  It removes a weird edge case and increases the security posture of the system.  Maintaining a small patch to better ensure a VHD isn't mounted read-write except when an admin/owner explicitly wants it is an acceptable price in my opinion, given how disastrous user-error can be.